### PR TITLE
Check that texture width and height is valid before creating texture

### LINKED ISSUE #8578: Creating a texture with width or height smaller than 1 will crash the engine
**Describe the bug (REQUIRED)**
It is possible to crash the engine when calling `resource.create_texture()` with a bad config.

**To Reproduce (REQUIRED)**
1. Call `resource.create_texture()` with a width and/or height less than 1
2. Observe crash

**Expected behavior (REQUIRED)**
It should not crash

**Defold version (REQUIRED):**
  - Version 1.6.4

**Platforms (REQUIRED):**
 - Platforms: macOS, but most likely all

**Logs (OPTIONAL):**

```
Assertion failed: (resource_hash), function Release, file resource.cpp, line 1263.
INFO:CRASH: Successfully wrote Crashdump to file: /Users/bjornritzl/Library/Application Support/Defold/_crash
ERROR:CRASH: CALL STACK:
ERROR:CRASH: 0   dmengine                            0x0000000104f19304 _ZN7dmCrash7OnCrashEi + 104
ERROR:CRASH: 1   libsystem_platform.dylib            0x00000001869f5a24 _sigtramp + 56
ERROR:CRASH: 2   libsystem_pthread.dylib             0x00000001869c5cc0 pthread_kill + 288
ERROR:CRASH: 3   libsystem_c.dylib                   0x00000001868d1a40 abort + 180
ERROR:CRASH: 4   libsystem_c.dylib                   0x00000001868d0d30 err + 0
ERROR:CRASH: 5   dmengine                            0x0000000104f5d908 _ZN10dmResource7ReleaseEPNS_16SResourceFactoryEPv.cold.2 + 0
ERROR:CRASH: 6   dmengine                            0x0000000104d050ac _ZN10dmResource7ReleaseEPNS_16SResourceFactoryEPv + 144
ERROR:CRASH: 7   dmengine                            0x0000000104d47cf8 _ZN12dmGameSystem15ResSpriteCreateERKN10dmResource20ResourceCreateParamsE + 140
ERROR:CRASH: 8   dmengine                            0x0000000104d0c91c _ZN10dmResourceL14CreateResourceEPNS_17ResourcePreloaderEPNS_14PreloadRequestEPvj + 212
ERROR:CRASH: 9   dmengine                            0x0000000104d0c5b8 _ZN10dmResourceL23PreloaderTryPruneParentEPNS_17ResourcePreloaderEPNS_14PreloadRequestE + 148
ERROR:CRASH: 10  dmengine                            0x0000000104d0be04 _ZN10dmResourceL22PreloaderUpdateOneItemEPNS_17ResourcePreloaderEs + 944
ERROR:CRASH: 11  dmengine                            0x0000000104d0bd38 _ZN10dmResourceL22PreloaderUpdateOneItemEPNS_17ResourcePreloaderEs + 740
ERROR:CRASH: 12  dmengine                            0x0000000104d0bd38 _ZN10dmResourceL22PreloaderUpdateOneItemEPNS_17ResourcePreloaderEs + 740
ERROR:CRASH: 13  dmengine                            0x0000000104d0bd38 _ZN10dmResourceL22PreloaderUpdateOneItemEPNS_17ResourcePreloaderEs + 740
ERROR:CRASH: 14  dmengine                            0x0000000104d0bd38 _ZN10dmResourceL22PreloaderUpdateOneItemEPNS_17ResourcePreloaderEs + 740
ERROR:CRASH: 15  dmengine                            0x0000000104d0b8b8 _ZN10dmResource15UpdatePreloaderEPNS_17ResourcePreloaderEPFbPKNS_31PreloaderCompleteCallbackParamsEEPS2_j + 292
ERROR:CRASH: 16  dmengine                            0x0000000104d14428 _ZN12dmGameSystem25CompCollectionProxyUpdateERKN12dmGameObject22ComponentsUpdateParamsERNS0_22ComponentsUpdateResultE + 212
ERROR:CRASH: 17  dmengine                            0x0000000104cedfb4 _ZN12dmGameObject6UpdateEPNS_16CollectionHandleEPKNS_13UpdateContextE + 448
ERROR:CRASH: 18  dmengine                            0x0000000104cda2e8 _ZN8dmEngine4StepEPNS_6EngineE + 892
ERROR:CRASH: 19  dmengine                            0x0000000104cda8c8 _Z14dmEngineUpdatePN8dmEngine6EngineE + 32
ERROR:CRASH: 20  dmengine                            0x0000000104cdb4a0 _ZN8dmEngine7RunLoopEPKNS_13RunLoopParamsE + 104
ERROR:CRASH: 21  dmengine                            0x0000000104cdb388 _Z11engine_mainiPPc + 88
ERROR:CRASH: 22  dyld                                0x00000001866450e0 start + 2360
```

### PR DESCRIPTION
**Pull request by britzl**
Calling `resource.create_texture()` with a width and/or height of zero crashed the engine. This change will check that the width and height are valid and throw a Lua error otherwise.

Fixes #8578 

## PR checklist

* [ ] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [x] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/gamesys/src/gamesys/scripts/script_resource.cpp [hunk 1]
```diff
@@ -790,10 +790,15 @@ static int CreateTexture(lua_State* L)
     luaL_checktype(L, 2, LUA_TTABLE);
     dmGraphics::TextureType type     = (dmGraphics::TextureType) CheckTableInteger(L, 2, "type");
     dmGraphics::TextureFormat format = (dmGraphics::TextureFormat) CheckTableInteger(L, 2, "format");
-    uint32_t width                   = (uint32_t) CheckTableInteger(L, 2, "width");
-    uint32_t height                  = (uint32_t) CheckTableInteger(L, 2, "height");
+    int width                        = CheckTableInteger(L, 2, "width");
+    int height                       = CheckTableInteger(L, 2, "height");
     uint32_t max_mipmaps             = (uint32_t) CheckTableInteger(L, 2, "max_mipmaps", 0);
 
+    if ((width < 1) || (height < 1))
```
Comment (Jhonnyg) [inline engine/gamesys/src/gamesys/scripts/script_resource.cpp:797]: redundant parentheses (nitpick)

[the remark this answers is not in the document]
Comment (Jhonnyg) [reply engine/gamesys/src/gamesys/scripts/script_resource.cpp:797]: I think ints is fine

File: engine/gamesys/src/gamesys/test/resource/create_texture.script [hunk 2]
```diff
@@ -51,6 +58,12 @@ function test_simple()
     go.set("#model", "texture0", t_id)
 end
 
+function test_invalid()
+    assert_error(function()
+        resource.create_texture("/test_invalid.texturec", tparams_invalid)
+    end)
+end
```
Comment (Jhonnyg) [inline engine/gamesys/src/gamesys/test/resource/create_texture.script:65]: 🤙

**Review by Jhonnyg — DISMISSED** [review]
I think it looks good, I would suggest adding a line in the docstring to say that the width and height must be > 0. What about set_texture? Do we need to do something for that as well?

**Comment by britzl** [comment]
> I think it looks good, I would suggest adding a line in the docstring to say that the width and height must be > 0. What about set_texture? Do we need to do something for that as well?

Good point about set_texture. Let me check!

**Comment by britzl** [comment]
> I think it looks good, I would suggest adding a line in the docstring to say that the width and height must be > 0. What about set_texture? Do we need to do something for that as well?

No, that seems to work.

**Review by Jhonnyg — APPROVED** [review]

**Pull request merged**